github-pullrequest-plugin/src/main/java/org/jenkinsci/plugins/github/pullrequest/trigger/JobRunnerForCause.java

```diff
@@ -137,8 +139,22 @@ private static boolean cancelQueuedBuildByPrNumber(final int id) {
                 NUMBER.param(String.valueOf(cause.getNumber()))
         ));
         GitHubPRBadgeAction gitHubPRBadgeAction = new GitHubPRBadgeAction(cause);
-        //TODO no way to get quietPeriod, so temporary ignore it
-        return asParameterizedJobMixIn(job).scheduleBuild2(0, new CauseAction(cause), new ParametersAction(values),
+
+        ParameterizedJobMixIn parameterizedJobMixIn = asParameterizedJobMixIn(job);
+
+        int quietPeriod = 0;
+
+        // TODO: figure out a better way to do this
+        try {
+            Object mixinJob = FieldUtils.readField(parameterizedJobMixIn, "val$job", true);
+            ParameterizedJobMixIn.ParameterizedJob parameterizedJob = (ParameterizedJobMixIn.ParameterizedJob) mixinJob;
+            quietPeriod = parameterizedJob.getQuietPeriod();
+            LOGGER.debug("Got quiet period from the project: {}", quietPeriod);
+        } catch (IllegalAccessException e) {
+            LOGGER.error("Couldn't get quiet period ({})", e.getMessage(), e);
```

Comment:
no, i mean print variable) in parallel working with puppet so head bit full of different lang syntaxes.